src/main/java/com/pingcap/tikv/region/TiRegion.java

```diff
@@ -128,15 +128,15 @@ public boolean contains(ByteString key) {
   }
 
   public boolean isValid() {
-    return peer != null && meta.hasStartKey() && meta.hasEndKey();
+    return peer != null && meta != null && hasEndKey() && hasEndKey();
   }
 
   boolean hasStartKey() {
-    return meta.hasStartKey();
+    return meta.getStartKey() != null;
```

Comment:
If it's not optional, doesn't protobuf stub generate a default?
So this test might be useless. Please confirm.

Reply:
In protobuf3, all fields have default value of 0, so this test seems useless.

Reply:
https://developers.google.com/protocol-buffers/docs/proto3#default